src/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,367 @@
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background-color: rgba(0, 0, 0, 0.4);
+  animation: dialog-backdrop-appear ${ANIMATION_DURATION} ${get('animation.easeOutCubic')};
+
+  @keyframes dialog-backdrop-appear {
+    0% {
+      opacity: 0;
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+`
+
+const heightMap = {
+  sm: '480px',
+  md: '640px',
+  auto: 'auto'
+}
+
+const widthMap = {
+  sm: '296px',
+  md: '320px',
+  lg: '480px',
+  xl: '640px',
+  auto: 'auto'
+}
+
+interface StyledDialogProps {
+  width?: keyof typeof widthMap
+  height?: keyof typeof heightMap
+}
+
+const StyledDialog = styled.div<StyledDialogProps & SystemCommonProps & SystemPositionProps & SxProp>`
+  display: flex;
+  flex-direction: column;
+  background-color: ${get('colors.bg.overlay')};
+  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.12), 0px 8px 24px rgba(149, 157, 165, 0.2);
+  min-width: 296px;
+  max-width: min(640px, 100vw);
+  min-height: 480px;
+  max-height: min(640px, 100vh);
+  width: ${props => widthMap[props.width ?? 'auto']};
+  height: ${props => heightMap[props.height ?? 'auto']};
+  border-radius: 12px;
+  overflow: hidden;
+  opacity: 1;
+  animation: overlay--dialog-appear ${ANIMATION_DURATION} ${get('animation.easeOutCubic')};
+
+  @keyframes overlay--dialog-appear {
+    0% {
+      opacity: 0;
+      transform: scale(0.5);
+    }
+    100% {
+      opacity: 1;
+      transform: scale(1);
+    }
+  }
+
+  ${COMMON};
+  ${POSITION};
+  ${sx};
+`
+
+const DefaultHeader: React.FC<DialogHeaderProps> = ({dialogLabelId, title, subtitle, dialogDescriptionId, onClose}) => {
+  const onCloseClick = useCallback(() => {
+    onClose('close-button')
+  }, [onClose])
+  return (
+    <Dialog.Header>
+      <Flex>
+        <Flex pt={2} flexDirection="column" flexGrow={1}>
+          <Dialog.Title id={dialogLabelId}>{title ?? 'Dialog'}</Dialog.Title>
+          {subtitle && <Dialog.Subtitle id={dialogDescriptionId}>{subtitle}</Dialog.Subtitle>}
+        </Flex>
+        <Dialog.CloseButton onClose={onCloseClick} />
+      </Flex>
+    </Dialog.Header>
+  )
+}
+const DefaultBody: React.FC<DialogProps> = ({children}) => {
+  return <Dialog.Body>{children}</Dialog.Body>
+}
+const DefaultFooter: React.FC<DialogProps> = ({footerButtons}) => {
+  const {containerRef: footerRef} = useFocusZone({
+    bindKeys: FocusKeys.ArrowHorizontal | FocusKeys.Tab,
+    focusInStrategy: 'closest'
+  })
+  return footerButtons ? (
+    <Dialog.Footer ref={footerRef as React.RefObject<HTMLDivElement>}>
+      <Dialog.Buttons buttons={footerButtons} />
+    </Dialog.Footer>
+  ) : null
+}
+
+const _Dialog = React.forwardRef<HTMLElement, React.PropsWithChildren<DialogProps>>((props, forwardedRef) => {
+  const {title = 'Dialog', subtitle = '', renderHeader, renderBody, renderFooter, onClose, role = 'dialog'} = props
+  const dialogLabelId = uniqueId()
+  const dialogDescriptionId = uniqueId()
```

Comment:
These should probably be prefixed with `dialog-` or something like that just to avoid collision elsewhere.  Also, just realized these are never memoized in DropdownMenu or ActionMenu.  Probably something we should consider changing so that we don't change the ids in the DOM on every re-render.

Reply:
How would this collide?

Reply:
Oh, I didn't realize that `uniqueId` already prepends `__primer_id_`.  I thought it was just a number.  Disregard!